spec/PeriodicBackgroundSync-index.bs

```diff
@@ -0,0 +1,95 @@
+        text: waiting worker; url: dfn-waiting-worker
+    type: interface
+        text: ExtendableEvent; url: extendable-event-interface
+        text: ExtendableEventInit; url: extendable-event-init-dictionary
+        text: ServiceWorkerGlobalScope; url: service-worker-global-scope-interface
+        text: ServiceWorkerRegistration; url: service-worker-registration-interface
+</pre>
+
+Introduction {#intro}
+=====================
+
+  <em>This section is non-normative.</em>
+
+  Web Applications often run in environments with unreliable networks (e.g., mobile phones) and unknown lifetimes (the browser might be killed or the user might navigate away). This makes it difficult to synchronize client data from web apps (such as photo uploads, document changes, or composed emails) with servers. It also makes it difficult to synchronize updated content on the servers with the web app. If the browser closes or the user navigates away before synchronization can complete, the app must wait until the user revisits the page to try again. This specification provides a new onperiodicsync [=service worker=] event which can fire [=in the background=] so that synchronization attempts can continue despite adverse conditions when initially requested. This API is intended to reduce the time between content creation and content synchronization between the server and the web app.
+
+  As this API relies on service workers, functionality provided by this API is only available in a [=secure context=].
+  ## Example ## {#example}
+  Requesting a [=periodic Background Sync opportunity=] at a daily frequency from a [=browsing context=]:
+
+  <pre class="lang-js">
+    async function getLatestNews() {
+      const registration = await navigator.serviceWorker.ready;
+      try {
+        await registration.periodicSync.register('get-latest-news', {
+          minInterval: 24 * 60 * 60 * 1000,
+        });
+      } catch (error) {
+        console.log('Periodic Sync could not be registered!');
+      }
+    }
+  </pre>
+
+  Reacting to a periodicsync event within a [=service worker=]:
+
+  <pre class="lang-js">
+    self.addEventListener('periodicsync', event => {
+      event.waitUntil(fetchAndCacheLatestNews());
+    });
+  </pre>
+
+  In the above example <code>fetchAndCacheLatestNews</code> is a developer-defined function.
+
+Concepts {#concepts}
+========================
+
+The periodicsync event is considered to run <dfn>in the background</dfn> if no [=service worker clients=] whose [=frame type=] is top-level or auxiliary exist for the origin of the corresponding service worker registration.
+
+The user agent is considered to be <dfn>online</dfn> if the user agent has established a network connection. A user agent MAY use a stricter definition of being [=online=]. Such a stricter definition MAY take into account the particular [=service worker=] or origin a [=periodicsync registration=] is associated with.
```

Comment:
Surely there's a definition of _offline_ elsewhere that we can re-use? It's easy to rabbit hole on what it means to be "online" that I think you'll want to avoid.

Reply:
BackgroundSync is using the same definition of online. I looked through several other specifications, and couldn't find a nice definition of online I could reuse. Please could you suggest one?
Alternately, I can make this a non-definition.

Reply:
Then refer to that one? If something already exists then we shouldn't redefine it.

Reply:
Dito re: the definition of `in the background`, best to not redefine it.